pytorch_lightning/utilities/parsing.py

```diff
@@ -197,72 +197,70 @@ def __repr__(self):
         return out
 
 
-def lightning_hasattr(model, attribute):
-    """ Special hasattr for lightning. Checks for attribute in model namespace,
-        the old hparams namespace/dict, and the datamodule. """
+def lightning_get_all_attr_holders(model, attribute):
+    """ Special attribute finding for lightning.  Gets all of the objects or dicts that holds attribute.
+            Checks for attribute in model namespace, the old hparams namespace/dict, and the datamodule. """
     trainer = getattr(model, 'trainer', None)
 
-    attr = False
+    holders = []
+
     # Check if attribute in model
     if hasattr(model, attribute):
-        attr = True
+        holders.append(model)
+
     # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
+    if hasattr(model, 'hparams'):
+        if attribute in model.hparams:
+            holders.append(model.hparams)
+
     # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    if not attr and trainer is not None:
-        attr = hasattr(trainer.datamodule, attribute)
+    if trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
```

Comment:
`datamodule` is also quite new
```suggestion
    if trainer is not None and hasattr(trainer, 'datamodule') is not None and hasattr(trainer.datamodule, attribute):
```

Reply:
I'm not against being extra-safe but this should not fail. Unless a user is using this code with their own `Trainer` that does not inherit PL's

Reply:
@Borda I think we don't need this, since we only expect this code to be used with the current version of the trainer and not some old. 

Plus, if you want to add this, instead of `hasattr(trainer, 'datamodule')` you'd have to do `getattr(trainer, 'datamodule', None) sinde `hasattr` only return True or False.